contrib/babelfishpg_tsql/src/fts_parser.y

```diff
@@ -123,24 +122,34 @@ char* translate_simple_term(const char* inputStr) {
     char* trimmedInputStr;
 
     // Check for empty input
-    if (inputStr == NULL) {
+    if (!inputStr || !(inputLength = strlen(inputStr))) {
         return NULL;
```

Comment:
Have you validated that the consumers of this function are capable of handling NULL values? I just checked, and it looks like SQL Server actually just returns an error on a NULL or empty input, so we should probably just copy that behavior
```
1> select * from t1 where contains(data, '')
2> go
Msg 7645, Level 15, State 1, Server EC2AMAZ-R8NSHGO, Line 1
Null or empty full-text predicate.
```

Reply:
Will check on this.

Reply:
Unless you are absolutely sure that the consumers of this function can handle NULL, this should be replaced with an ereport()